### ISSUE TITLE
UI overhaul needed

### ISSUE THREAD

**Issue by jwr1** [jwr1/interstellar#60]
### Description

Interstellar desperately needs a UI overhaul; some parts of the UI are overly cramped, some parts are unintuitive, and many parts the visual aspect could be improved as they just don't look good. I'm no graphic/ui designer, so if there's anyone that would like to help, please do so, other wise, I'll need to take inspiration from other fediverse/forum apps that actually do look good.

Several people have mentioned wanting a "compact" mode. I'm not exactly sure what they are thinking of (especially since we already have a "compact preview" option), but it's something to keep in mind (maybe less padding between cards?).

Related issues:

- [ ] #26
- [ ] #59

**Comment by AndrewMathis13** [comment]
I can play around with things this weekend and try to rework the interface to simplify it a bit. I don't know enough about programming to actually write code for any of the changes, but I can definitely provide mockups with dimensions for anyone else who has the time and technical know-how to implement them.

**Comment by jwr1** [comment]
That would be great! Any help is welcome :)

I just want to put this here, but I think the UI design from the [vucuit app](https://codeberg.org/vuecuit/vuecuit/src/branch/master/SCREENSHOTS.md) looks nice. There are several Lemmy apps we could take inspiration from as well.

**Comment by dalz** [comment]
Regarding the compact view, compare Interstellar:

![Screenshot_20240913-000449](https://github.com/user-attachments/assets/79c95fd1-d488-4e2c-9539-00a57524bd4e)

with Redreader:

![Screenshot_20240913-000401](https://github.com/user-attachments/assets/ae730907-f106-4980-bc5a-f7720845de4d)

Redreader fits 7 posts with full titles, Interstellar 2 with truncated titles.

**Comment by jwr1** [comment]
@dalz, you're right. I see what you mean. Even setting the Media Position setting to the right side (instead of top) doesn't help.

**Comment by AndrewMathis13** [comment]
That's kind of where I was going too. I played around with it a little, but I couldn't find a way to reduce the size of the standard card enough to justify a compact mode as-is. So, I decided to instead just strip it down to the basics and rearrange the layout from there, and I think this works a lot better. 

<img width="1116" alt="Interstellar-App-New-Card-Design-v3" src="https://github.com/user-attachments/assets/358a2913-8c0a-4f16-8f2a-dc3fa7579115">

**Comment by dalz** [comment]
@AndrewMathis13 looks great! I'd suggest removing the space around the cards too, or at least have an option to do so, to not waste any space.

I'd like to note also that the top and bottom bars take up a lot of space and do not go away while scrolling. IMO they should aleays hide on scroll, even on the standard layout. It would be nice if they could be removed, maybe moving their content to the floating button or sidebar, but it could be low priority.

**Comment by jwr1** [comment]
Agreed, @AndrewMathis13, that looks great! When I get a chance, I'll see what I can do to replicate it in code.

@dalz, I know this wouldn't be the same as hiding the top part, but you can at least hide the "tabs" part of the top bar in the settings, so that it looks like this instead:

![image](https://github.com/user-attachments/assets/fa8a4011-3b5d-486d-8bfe-ffd8d51b69a8)

If you want to do that, go to settings, then Actions and Defaults, then change the "Set Filter" location to be anything else.

**Comment by AndrewMathis13** [comment]
Great! Below I'll include the measurements I was working with for both the page itself and both styles of cards to hopefully make implementing them easier.

Also, the search page still needs a few more adjustments, but after that I'll start working on the profile and settings pages this weekend.

<img width="1476" alt="Interstellar-Home-Page-Design-v1" src="https://github.com/user-attachments/assets/9f703af4-1f60-4b3a-9ad5-7717fc916d0e">

**Comment by AndrewMathis13** [comment]
It took me a little longer than I originally anticipated, but I've finally finished the explore, notifications, and chat pages. I'm still working on the settings page, and the profile page needs a few more tweaks, but then I can work on finishing up the other miscellaneous pages as well as the compact mode and a better optimized layout for large screen devices.

For the explore page, I thought it would be better to condense it all down into one page, and then provide the user with options to filter through the different search types. If there is some way to merge all the search functions into one bar with one singular feed of results, with the option to then filter by result type if desired similar to how search works on Mastodon apps, I think that would be optimal. However, if that's not possible, then we can just remove the "all" filter chip, and just use the other buttons to switch between the different search types, but ideally still keeping everything on the same page.

No big changes for the notifications page, just a few touchups here and there. However, I did kind of make a pretty big overhaul to the chat page lol. But the redesigned chat page isn't functionally any different than it is currently, so, with the exception of the updated text input field, it should only be visual changes. But if the updated design for the text input field, or honestly even the whole chat page redesign, ends up being too difficult to implement in code, I can definitely work around it and try to come up with something else instead.

<img width="2009" alt="Interstellar-Explore-Page-Design-v1" src="https://github.com/user-attachments/assets/6d1e60d7-f107-4fca-9196-714665fd162d">

<img width="1993" alt="Interstellar-Notification-Page-Design-v1" src="https://github.com/user-attachments/assets/e435f117-1169-4a55-a573-a0ec7ec15e57">

<img width="2607" alt="Interstellar-Messages-Page-Design-v1" src="https://github.com/user-attachments/assets/8758bb60-2a13-4277-addd-38166fe307d4">

**Comment by jwr1** [comment]
Wow, that looks amazing! I think it totally makes sense to condense the Explore page; a lot of the filtering/inputs/interface was the same between the tabs anyway. I also like the design overhaul for the messaging page; it makes it a lot more consistent with other chat interfaces.

Unfortunately, for the Explore changes, you actually have to search something first before it will show anything for the "All Search", whereas for the "Magazine Search", "People Search", and "Domain Search", you actually don't have to search anything and it will show you a full list of magazines, people, or domains. So maybe we could still have the "All", "Magazine", "People", and "Domain" filters as you have in the design, but we could have the "Magazine" filter enabled by default, and then if someone switches to the "All" filter, we could show a message saying they have to enter something in the search field first.

Overall, I love the designs, and I can't wait to implement them! I just need to see when I actually have time to do so.

**Comment by jwr1** [comment]
I mostly got the chat/messaging interface done. I did add a few things, such as the timeline for date & time, and grouping messages from the same person. I still need to work on the input field though; I'm just not exactly sure what to do with it, since the current version has a lot going on in it. I could leave it as is, or I could make it so it's just a simple text field that expands into the whole editor when a button is clicked on (like how it is in the design).

Anyway, here are some screenshots:

![image](https://github.com/user-attachments/assets/85abe090-428f-4b53-9d07-fab1f2793cbc)
![image](https://github.com/user-attachments/assets/6d2cf342-b25c-47bd-b51b-4c56e1185679)
![image](https://github.com/user-attachments/assets/d81f8892-5848-41f8-bffd-e55638ab2627)

**Comment by jwr1** [comment]
Here's the new notifications screen:

![image](https://github.com/user-attachments/assets/9ca5ea6f-d05c-4676-98e5-f5d92293aa4e)

**Comment by AndrewMathis13** [comment]
Wow, it's looking great. Hopefully it hasn't been too much trouble to implement. Also, I think adding timeline labels to the chat interface was a great idea!

I've also (mostly) finished with the user's profile page, though I do still need to make a few more tweaks to the page when viewing someone else's profile.

<img width="2116" alt="Interstellar-Profile-Page-Design-v1" src="https://github.com/user-attachments/assets/eaea1efd-ac05-4e97-bdd9-d76a29750960">

**Comment by AndrewMathis13** [comment]
Okay, the Settings page is done. I took inspiration from the Moshidon Mastodon client and regrouped the various settings into six different groups and gave each of them their own sub-page. All of the settings options have the exact same design specifications, 12dp top and bottom with 16dp spacing on the left and right and in-between elements, unless otherwise noted in the images below.

While I was redesigning the settings page I thought of some functional changes and some additional settings options that other fediverse clients have that I thought Interstellar could also benefit from:

- Add option to change where web links open.
- Add reminder to add alt-text to posts.
- Option to disable gifs from auto playing.
- Option to ask before unfollowing someone or a magazine, before boosting a post, or before deleting a post.
- Option to automatically load new posts without having to refresh the feed.
- Option for "See new posts" button when new posts are loaded on the feed.
- Option to disable haptic feedback on the device.
- "Use Dynamic Color" and "Color Scheme" can be combined into "Color palette."
- Option to always reveal content warnings.
- Option to hide/show media marked as sensitive.
- Accessibility feature to reduce animations for app interactions.
- Option to disable the ability to swipe between tabs.
- Option to choose which type of notifications the app will notify you about.



<img width="2118" alt="Interstellar-Settings-Page-1-Design-v1" src="https://github.com/user-attachments/assets/53426597-12d0-43e3-b0ed-6f460e75bd02">

<img width="2680" alt="Interstellar-Settings-Page-2-Design-v1" src="https://github.com/user-attachments/assets/38e36410-ffd9-421f-a2ed-f188f13d8916">

**Comment by AndrewMathis13** [comment]
Thanks, I’ve really enjoyed working on the redesign. Also it’s totally not too much feedback, definitely keep it coming. It only makes the designs better.

- I had completely forgotten about Notification Channels in Android! So then yes, I think I would be better to consolidate all the notification settings into the Settings page and then have the app only modify settings via the API, and then the user can filter out different kinds of notifications from Android system settings if desired. 
- Initially they were supposed to be different. “Always reveal” was for expanding the content warning on the thread or post, while “Cover up media” was to blur or hide images even if the text content of the post was visible. In hindsight though, they could probably be combined into one toggle. But if we instead want to keep them as separate options, then they should probably be renamed to avoid confusion.
- That’s a great point. That would be better placed on the Behaviors page, maybe under the toggle for animated gifs?
- It could be part of the same thing, but it’s whether or not you want Interstellar to automatically add new posts to the top of the feed as they’re published versus the user pulling down to refresh to load the new posts. So the top of the feed wouldn’t necessarily have a “show new posts” button, but if it was combined with infinite scrolling, then it would at the bottom of the feed or page if the feature is disabled. 
- As a part of the “automatically load new posts” feature in a lot of Mastodon clients, a button comes down when there are new posts available at the top of the feed, and if clicked, the button takes the user up to the top to view the new posts.

![Screenshot_20241011-180934-edit](https://github.com/user-attachments/assets/52c0012f-f1d8-402b-a8d3-aa366020e418)

- We could maybe just rename “Feed Actions” to “Filters” and then move the “Advanced” subsection out and into the “Display” page instead. It would have to be renamed obviously, but if we added another divider above “Always show instance” we could place the advanced feed actions sub-page there in-between the dividers. I guess my question then is would we also want to move the “Feed Defaults” under the “Display” page as a sub-page as well? Though as a follow up, should they go under “Display”, or do you think it makes more sense for them to go under “Behavior?”
- I think that’s a good idea. As for the “Get notifications from” it was initially going to be from the preset options of Anyone, Followers, Mutuals, Following, or No one. Though the API probably won’t support that either, but I do like the idea of a deny list of accounts a user doesn’t want to be notified for. The notification should probably still be shown on the Notifications page, but Interstellar wouldn’t notify the user. I also like the idea of moving the Register Push toggle to the main notifications settings page.
- Yeah, several Mastodon clients I’ve used on Android group filters together like this, and I may be a little biased, but I think it’s super helpful and additionally it keeps UI a little cleaner too. Personally I haven’t set a duration limit on any filters yet, but I think it could be beneficial if a user just wants to take a break from a specific topic for a while, or wants to avoid seeing any spoilers for a film or show before they’ve had a chance to watch it. But the official Mastodon Android client plus a few others that I’ve tried also have the duration limit for filters. I do think adding the option to switch to regex is a good idea, but I think we’d want to put the toggle on the page for an individual filter group, so users can have more granular control over how they want the filters to apply. 
- That’s fair. There are a few options we could implement additionally. We could replace the account icon in the bottom navigation with the profile image of the active account, again like a lot of Mastodon clients do. If we did that, then add a long press to the profile image on the nav bar to invoke the account switcher outside of the settings page. Or we could replace the account icon in settings with the user’s profile image. If you don’t like either of those ideas, then I could try messing around with the main settings page and try to add the profile image, username, and handle to the page instead of the button. Or potentially, I could try adding the “Log Out” button to the account switcher instead of it being a separate option. 
- Okay, well in that case we can just remove the about section completely and I can put the date joined elsewhere on the main profile card. For the “What’s in a handle?” popup, it should show up if you select either part of a user’s handle, and then you can copy the account address by selecting the box in the popup with the user’s full handle. The popup should be accessible on all profiles, the only difference being  the text “Your handle” should be replaced with “Their handle” when viewing someone else’s profile.
- And as for the posts, yes it would have to be the normal card. However, I am considering changing the layout a little when viewing a full post and comments over just displaying the same card a second time. Though I still need to figure out how I’m going to tackle displaying threaded comments under the post, so I’m not a hundred percent certain yet how it’s all going to look.

Additionally, I would love to design a welcome screen for Interstellar. I still need to finish the expanded posts/comments view, make a few more changes to the profile page when viewing someone else’s profile, update the magazine page first, and then, as long as I’m not forgetting something else, I can start on the welcome screen. I do also need to make a few tweaks to the whole layout for large screen devices, but I’ll do that last.

Again, thanks for all of the feedback, it really helps. Also, we can totally move this conversation over to Matrix if that’s easier for you.

**Comment by jwr1** [comment]
Alright, that all sounds good to me :+1:.

I just made a new matrix room dedicated to development that we could chat in: https://matrix.to/#/#interstellar-development:matrix.org. If I have any small questions while implementing things, I'll just ask you there if that's fine.

**Comment by melroy89** [comment]
@AndrewMathis13 wow again very impressive! Your work doesn't go unnoticed!

**Comment by jwr1** [comment]
> @AndrewMathis13 wow again very impressive! Your work doesn't go unnoticed!

Yes indeed! It's pretty amazing.

**Comment by Dark-Thoughts** [comment]
> That's kind of where I was going too. I played around with it a little, but I couldn't find a way to reduce the size of the standard card enough to justify a compact mode as-is. So, I decided to instead just strip it down to the basics and rearrange the layout from there, and I think this works a lot better.
> <img alt="Interstellar-App-New-Card-Design-v3" width="1116" src="https://private-user-images.githubusercontent.com/95931220/367118718-358a2913-8c0a-4f16-8f2a-dc3fa7579115.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyNDY1NzQsIm5iZiI6MTczMzI0NjI3NCwicGF0aCI6Ii85NTkzMTIyMC8zNjcxMTg3MTgtMzU4YTI5MTMtOGMwYS00ZjE2LThmMmEtZGMzZmE3NTc5MTE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDE3MTc1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5NzQ2ODY2ZjI3ZGQ1OGVkN2Y2YmI0M2VjODczZTYzYzJiMDVhZDM3YWQwMTQ0ZmE4YjVkMTBlMjFlNDlkZTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C3J5qQzxyhSYKU5tIay_x-wFK9VdqbaVJ8ab0e2sZZI">

I don't if that's still the current design for the compact view, but personally I really don't like wasting so much space for padding and frames, especially when it means that we shrink the thumbnails down unnecessarily. [If we look at mbin's desktop compact mode, we can see that the thumbnails pretty much fill out its entire space in each row.](https://i.imgur.com/M41e6Bn.png)

@dalz You can change the function of the titles to open the comments while the thumbnail / rest of the row opens the linked content, which means you can remove the unnecessary comment button on the right as well.

**Comment by jwr1** [comment]
> I don't if that's still the current design for the compact view, but personally I really don't like wasting so much space for padding and frames, especially when it means that we shrink the thumbnails down unnecessarily. There's empty space within the frames of the different rows, and then again even more empty space between the different rows itself. [If we look at mbin's desktop compact mode, we can see that the thumbnails pretty much fill out its entire space in each row.](https://i.imgur.com/M41e6Bn.png) I'd prefer it even more if the rows were also touching and just separate with a line instead.

Yeah, I was actually planning on doing it that way (with no space between and just a divider) because I know the majority of people who would use compact mode would want it like that.

**Comment by jwr1** [comment]
@Dark-Thoughts, here's what I have for compact mode so far:

![](https://github.com/user-attachments/assets/7397ed7a-6224-42ed-b8dd-d7cc79abf898)

**Comment by Dark-Thoughts** [comment]
Yes, definitely better. I'd just remove the rounded edges on the thumbnail and the empty space around them, have them use that space fully from top to bottom.

**Comment by Dark-Thoughts** [comment]
Yesssssss!
